DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/ViewTreeSnapshotBuilder.swift

```diff
@@ -45,25 +59,36 @@ internal struct ViewTreeSnapshotBuilder {
 extension ViewTreeSnapshotBuilder {
     init() {
         self.init(
-            viewTreeRecorder: ViewTreeRecorder(nodeRecorders: defaultNodeRecorders),
-            idsGenerator: NodeIDGenerator(),
-            textObfuscator: TextObfuscator()
+            viewTreeRecorder: ViewTreeRecorder(nodeRecorders: createDefaultNodeRecorders()),
+            idsGenerator: NodeIDGenerator()
         )
     }
 }
 
 /// An arrays of default node recorders executed for the root view-tree hierarchy.
-internal let defaultNodeRecorders: [NodeRecorder] = [
-    UIViewRecorder(),
-    UILabelRecorder(),
-    UIImageViewRecorder(),
-    UITextFieldRecorder(),
-    UITextViewRecorder(),
-    UISwitchRecorder(),
-    UISliderRecorder(),
-    UISegmentRecorder(),
-    UIStepperRecorder(),
-    UINavigationBarRecorder(),
-    UITabBarRecorder(),
-    UIPickerViewRecorder(),
-]
+internal func createDefaultNodeRecorders() -> [NodeRecorder] {
+    let imageViewRecorder = UIImageViewRecorder()
+    imageViewRecorder.semanticsOverride = { imageView, _ in
+        let className = "\(type(of: imageView))"
+        // This gets effective on iOS 15.0+ which is the earliest version that displays some elements in popover views.
+        // Here we explicitly ignore the "shadow" effect applied to popover.
+        let isSystemShadow = className == "_UICutoutShadowView"
+        return isSystemShadow ? IgnoredElement(subtreeStrategy: .ignore) : nil
```

Comment:
It almost sounds like it should be default value of the `semanticsOverride` in the init.

We probably want to expand the coverage of non-essential images here as well.

Reply:
Sounds fair 👍, moved this to default value on `UIImageViewRecorder`.